server/models/user.js

```diff
@@ -0,0 +1,79 @@
+const mongoose = require('mongoose');
+const validator = require('validator');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+
+
+const userSchema = mongoose.Schema({
+    username: {
+        type: String,
+        required: true,
+        trim: true
+    },
+    email: {
+        type: String,
+        required: true,
+        unique: true,
+        lowercase: true,
+        validate: value => {
+            if (!validator.isEmail(value)) {
+                throw new Error({error: 'Invalid Email address'})
+            }
+        }
+    },
+    password: {
+        type: String,
+        required: true,
+        minlength:6,
+    }
+})
+
+
+// Hash the password before saving the user model
+userSchema.pre('save', async function (next) {
+    const user = this
+    const user_pw = user.password;
+    if (user.isModified('password')) {
+        user.password = await bcrypt.hash(user.password, 8)
+    }
+
+    // verify a hash
+    const test_hash = bcrypt.compareSync(user_pw, user.password);
+    if (test_hash){
+        console.log("hash verified");
+    } else {
+        console.log("hash failed");
+    }
+

```

Comment:
you shouldn't need to verify the password at this point, since you just used it to create the hash